common/src/main/java/tachyon/client/RawTableMasterClient.java

```diff
@@ -0,0 +1,177 @@
+      }
+    });
+  }
+
+  /**
+   * Get the {@link RawTableInfo} associated with the given id.
+   *
+   * @param id the id of the table
+   * @return the table info
+   * @throws IOException when operation fails
+   */
+  public synchronized RawTableInfo getClientRawTableInfo(final long id) throws IOException {
+    return retryRPC(new RpcCallable<RawTableInfo>() {
+      @Override
+      public RawTableInfo call() throws TException {
+        return mClient.getClientRawTableInfoById(id);
+      }
+    });
+  }
+
+  /**
+   * Get the {@link RawTableInfo} associated with the given path.
+   *
+   * @param path the path of the table
+   * @return the table info
+   * @throws IOException when operation fails
+   */
+  public synchronized RawTableInfo getClientRawTableInfo(final TachyonURI path) throws IOException {
+    return retryRPC(new RpcCallable<RawTableInfo>() {
+      @Override
+      public RawTableInfo call() throws TException {
+        return mClient.getClientRawTableInfoByPath(path.getPath());
+      }
+    });
+  }
+
+  /**
+   * Update the metadata of a table.
+   *
+   * @param tableId The id of the table to update
+   * @param metadata The new metadata to update the table with
+   * @throws IOException when the operation fails
+   */
+  public void updateRawTableMetadata(final long tableId, final ByteBuffer metadata)
+      throws IOException {
+    retryRPC(new RpcCallable<Void>() {
+      @Override
+      public Void call() throws TException {
+        mClient.updateRawTableMetadata(tableId, metadata);
+        return null;
+      }
+    });
+  }
+
+  // TODO(cc) {@link #RpcCallable} and {@link #retryPRC}  may be valuable to other clients too, but 
+  // let's limit its use to this class in this PR, may be replaced if a better solution is found.
+  /**
+   * The RPC to be executed in {@link #retryRPC(RpcCallable)}.
+   *
+   * @param <V> the return value of {@link #call()}
+   */
+  private interface RpcCallable<V> {
+    /**
+     * The task where RPC happens.
+     *
+     * @return RPC result
+     * @throws TException when any exception defined in thrift happens
+     */
+    V call() throws TException;
+  }
+
+  /**
+   * Tries to execute a RPC defined as a {@link RpcCallableWithPropagateTException}, if error
+   * happens in one execution, a reconnection will be tried through {@link #connect()} and the
+   * action will be re-executed.
+   *
+   * @param rpc the {@link RpcCallable} to be executed
+   * @param <V> type of return value of the RPC call
+   * @return the return value of the RPC call
+   * @throws IOException when retries exceeds {@link #RPC_MAX_NUM_RETRY} or {@link #close()} has
+   *                     been called before calling this method or during the retry
+   */
+  private <V> V retryRPC(RpcCallable<V> rpc) throws IOException {
+    int retry = 0;
+    while (!mClosed && (retry ++) <= RPC_MAX_NUM_RETRY) {
+      connect();
+      try {
+        return rpc.call();
+      } catch (TException e) {
```

Comment:
As per the exception handling proposal, the Thrift RPC calls should only throw TException and TachyonTException. We throw the TachyonTException to the caller who can take care of the unwrapping.